test-suite/src/test/groovy/io/micronaut/docs/server/endpoints/CspReportEndpoint.groovy

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright 2017-2019 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.docs.server.endpoints
+
+//tag::import[]
+import io.micronaut.http.HttpStatus
+import io.micronaut.http.annotation.Body
+import io.micronaut.management.endpoint.annotation.Endpoint
+import io.micronaut.management.endpoint.annotation.Write
+//end::import[]
+
+//tag::classBegin[]
+@Endpoint(id = "report", defaultSensitive = false)
+class CspReportEndpoint {
```

Comment:
What is the purpose of this endpoint?

Reply:
Browsers could report violations if configured in the policy and the application could expose an endpoint where it gets captured, this is an example using Micronaut endpoints. This is just used as an example in the documentation.

Reply:
How would the data be captured?

Reply:
An endpoint is likely not necessary here given CSP is purely a web construct. Endpoints are for information that may need to be available outside of an HTTP environment

Reply:
Ok, I was thinking endpoints more of a management utility. I will update the example to not use an endpoint.

Reply:
If you think that the information may have usefulness to retrieve over JMX for example then you could suggest an endpoint, either way not a giant deal. I don't know enough about the use case to say for certain. Either way I think it would be important to show how the data is retrieved from the browser